lib/Doctrine/Common/Inflector/Inflector.php

```diff
@@ -148,7 +176,76 @@ public static function reset() : void
      */
     public static function rules(string $type, iterable $rules, bool $reset = false) : void
     {
-        throw new BadMethodCallException('Adding custom rules is no longer supported in Doctrine Inflector 2.0.');
+        @trigger_error(sprintf('The "%s" method is deprecated and will be dropped in Doctrine Inflector 3.0. Please update to the new Inflector API.', __METHOD__), E_USER_DEPRECATED);
+
+        if (self::$factory === null) {
+            self::$factory = self::createFactory();
+        }
+
+        self::$instance = null;
+
+        switch ($type) {
+            case 'singular':
+                self::$factory->withSingularRules(self::buildRuleset($rules), $reset);
+                break;
+            case 'plural':
+                self::$factory->withPluralRules(self::buildRuleset($rules), $reset);
+                break;
+            default:
+                throw new InvalidArgumentException(sprintf('Cannot define custom inflection rules for type "%s".', $type));
+        }
+    }
+
+    private static function buildRuleset(iterable $rules) : Ruleset
+    {
+        $regular = [];
+        $irregular = [];
+        $uninflected = [];
+
+        foreach ($rules as $rule => $pattern) {
+            if ( ! is_array($pattern)) {
+                continue;
+            }
+
+            switch ($rule) {
+                case 'uninflected':
+                    $uninflected = $pattern;
+                    break;
+                case 'irregular':
+                    $irregular = $pattern;
+                    break;
+                case 'rules':
+                    $regular = $pattern;
+                    break;
+            }
+
+            unset($rules[$rule]);
```

Comment:
`iterable` doesn't mean array access is possible. Users still can inject an iterator in `Inflector::rules()` that doesn't allow array access.

Reply:
Yep, that's what happens when you copy/paste legacy code. I'll refactor the loop and deal with these rules in the loop itself.

Reply:
Is there a sniff for coding styles that can catch an iterable that is used with array access?

Reply:
This file is not covered by the coding standard because it needs to be backwards compatible. You’d need a “lite” version of it.